components/authz-service/storage/v2/statement.go

```diff
@@ -21,6 +21,13 @@ type Statement struct {
 // validation around what a valid statement is in terms of our storage layer.
 // It also generates a new ID for our statement.
 func NewStatement(effect Effect, role string, projects, resources, actions []string) (Statement, error) {
+	if actions == nil {
+		actions = []string{}
```

Comment:
Do we ever want to allow empty actions? Is that valid? Or is it ok to have empty actions provided a role exists?

Reply:
a statement must have either actions or a role (it can also have both)